core/src/mender-artifact.c

```diff
@@ -1119,15 +1187,16 @@ artifact_read_data(mender_artifact_ctx_t *ctx, mender_err_t (*callback)(char *,
         }
 #endif /* CONFIG_MENDER_FULL_PARSE_ARTIFACT */
 
-        /* Invoke callback */
-        if (MENDER_OK
-            != (ret = callback(ctx->payloads.values[index].type,
-                               ctx->payloads.values[index].meta_data,
-                               strstr(ctx->file.name, ".tar") + strlen(".tar") + 1,
-                               ctx->file.size,
-                               ctx->input.data,
-                               ctx->file.index,
-                               length))) {
+        /* Invoke the download artifact callback */
+        ret = process_artifact_data_callback(ctx->payloads.values[index].type,
```

Comment:
From this line... such a simple change :smile:

Reply:
?

Reply:
It was meant a compliment. If you only look at this line your PR seems "so simple", yet the work is everywhere else to make this possible :rocket: